openlcs/packages/views.py

```diff
@@ -875,6 +875,35 @@ def retrieve(self, request, *args, **kwargs):
         """
         return super().retrieve(request, *args, **kwargs)
 
+    @action(methods=['post'], detail=False, url_path='filter')
+    def filter(self, request, *args, **kwargs):
+        filter_results = {'should_filter': False}
+        data = request.data
+        imported_components = Component.objects.filter(
+                name=data.get('name', ''),
+                version=data.get('version', ''),
+                release=data.get('release', ''),
+                arch=data.get('arch', ''),
+                type=data.get('type', ''),
+        )
+        serializer = ComponentSerializer(imported_components, many=True)
+        results = serializer.data
+
+        if results and len(results) == 1:
+            filter_results['should_filter'] = True
+            source_data = results[0].get('source')
+            if source_data is not None:
+                scan_flag = source_data.get('scan_flag')
```

Comment:
'scan_flag' checking could be removed from the duplicate checking IMO. PS: There is a 'state' in Source, I don't know what's its usage and where it is used.

Reply:
DONE